src/apps/hidden-hand/common/hidden-hand.rewards.contract-position-fetcher.ts

```diff
@@ -0,0 +1,138 @@
+import { Inject, NotImplementedException } from '@nestjs/common';
+import { Contract } from 'ethers';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { DefaultDataProps } from '~position/display.interface';
+import { CustomContractPositionTemplatePositionFetcher } from '~position/template/custom-contract-position.template.position-fetcher';
+import {
+  DefaultContractPositionDefinition,
+  GetTokenDefinitionsParams,
+  UnderlyingTokenDefinition,
+  GetDisplayPropsParams,
+} from '~position/template/contract-position.template.types';
+
+import { HiddenHandContractFactory, HiddenHandRewardDistributor } from '../contracts';
+import { HiddenHandRewardsResolver, HiddenHandRewardsDefinition, PROTOCOLS, REWARD_DISTRIBUTOR } from './hidden-hand.rewards-resolver';
+import { ContractPositionBalance } from '~position/position-balance.interface';
+import { drillBalance } from '~app-toolkit/helpers/drill-balance.helper';
+import _, { sumBy } from 'lodash';
+import { MetaType } from '~position/position.interface';
+import { ZERO_ADDRESS } from '~app-toolkit/constants/address';
+import { NETWORK_IDS } from '~types';
+
+export class HiddenHandRewardsContractPositionFetcher extends CustomContractPositionTemplatePositionFetcher<
```

Comment:
This class should be abstract because you don't want this class to live on it's own without any subclasses